plugins/in_proc/in_proc.c

```diff
@@ -158,66 +158,29 @@ static pid_t get_pid_from_procname_linux(struct flb_in_proc_config *ctx,
 static int configure(struct flb_in_proc_config *ctx,
                      struct flb_input_instance *in)
 {
-    const char *pval = NULL;
-
-    /* interval settings */
-    pval = flb_input_get_property("interval_sec", in);
-    if (pval != NULL && atoi(pval) >= 0) {
-        ctx->interval_sec = atoi(pval);
-    }
-    else {
-        ctx->interval_sec = DEFAULT_INTERVAL_SEC;
-    }
+    char *proc_name = NULL;
+    int ret;
 
-    pval = flb_input_get_property("interval_nsec", in);
-    if (pval != NULL && atoi(pval) >= 0) {
-        ctx->interval_nsec = atoi(pval);
-    }
-    else {
-        ctx->interval_nsec = DEFAULT_INTERVAL_NSEC;
+    /* Load the config map */
+    ret = flb_input_config_map_set(in, (void *)ctx);
+    if (ret == -1) {
+        return -1;
     }
-
+    
     if (ctx->interval_sec <= 0 && ctx->interval_nsec <= 0) {
         /* Illegal settings. Override them. */
-        ctx->interval_sec = DEFAULT_INTERVAL_SEC;
-        ctx->interval_nsec = DEFAULT_INTERVAL_NSEC;
+        ctx->interval_sec = atoi(DEFAULT_INTERVAL_SEC);
+        ctx->interval_nsec = atoi(DEFAULT_INTERVAL_NSEC);
     }
 
-    pval = flb_input_get_property("alert", in);
-    if (pval) {
-        if (strcasecmp(pval, "true") == 0 || strcasecmp(pval, "on") == 0) {
-            ctx->alert = FLB_TRUE;
-        }
-    }
-
-    pval = flb_input_get_property("mem", in);
-    if (pval) {
-        if (strcasecmp(pval, "true") == 0 || strcasecmp(pval, "on") == 0) {
-            ctx->mem = FLB_TRUE;
-        }
-        else if (strcasecmp(pval, "false") == 0 || strcasecmp(pval, "off") == 0) {
-            ctx->mem = FLB_FALSE;
-        }
-    }
-
-    pval = flb_input_get_property("fd", in);
-    if (pval) {
-        if (strcasecmp(pval, "true") == 0 || strcasecmp(pval, "on") == 0) {
-            ctx->fds = FLB_TRUE;
-        }
-        else if (strcasecmp(pval, "false") == 0 || strcasecmp(pval, "off") == 0) {
-            ctx->fds = FLB_FALSE;
-        }
-    }
-
-    pval = flb_input_get_property("proc_name", in);
-    if (pval) {
-        ctx->proc_name = (char*)flb_malloc(FLB_CMD_LEN);
+    if (ctx->proc_name != NULL && strcmp(ctx->proc_name, "") != 0) {
+        proc_name = (char*)flb_malloc(FLB_CMD_LEN);
         if (ctx->proc_name == NULL) {
             return -1;
         }
-        strncpy(ctx->proc_name, pval, FLB_CMD_LEN);
-        ctx->proc_name[FLB_CMD_LEN-1] = '\0';
+        strncpy(proc_name, ctx->proc_name, FLB_CMD_LEN);
+        proc_name[FLB_CMD_LEN-1] = '\0';
+        ctx->proc_name = proc_name;
```

Comment:
`ctx->proc_name` is already allocated by config_map framework.
I think this line overwrites it.

Reply:
I think `flb_free(ctx->proc_name);` of `in_proc_exit` should be removed since it will be freed by config_map framework.